src/cow-react/modules/limitOrders/hooks/useApplyLimitRate.ts

```diff
@@ -5,32 +5,39 @@ import { Field } from 'state/swap/actions'
 import { limitRateAtom } from '@cow/modules/limitOrders/state/limitRateAtom'
 import { Fraction } from '@uniswap/sdk-core'
 import { toFraction } from '@cow/modules/limitOrders/utils/toFraction'
+import { useLimitOrdersTradeState } from '@cow/modules/limitOrders/hooks/useLimitOrdersTradeState'
+import { adjustDecimals } from '@cow/modules/limitOrders/utils/adjustDecimals'
 
 // Applies rate to provided value which can be INPUT or OUTPUT
 export function useApplyLimitRate() {
   const { activeRate } = useAtomValue(limitRateAtom)
+  const { inputCurrency, outputCurrency } = useLimitOrdersTradeState()
 
   return useCallback(
-    (value: string | null, field: Field): string | null | undefined => {
-      if (!value || !Number(value) || !activeRate || activeRate.equalTo(0)) {
+    (value: string | null, field: Field): Fraction | null => {
+      if (!value || !Number(value) || !activeRate || activeRate.equalTo(0) || !inputCurrency || !outputCurrency) {
         return null
       }
 
-      let output: Fraction | null = null
-      const parsedValue = toFraction(value)
+      const { decimals: inputDecimals } = inputCurrency
+      const { decimals: outputDecimals } = outputCurrency
+
+      const parsedValue = adjustDecimals(
```

Comment:
It's a fix for trading currencies with different decimals:
https://github.com/cowprotocol/cowswap/pull/1654/commits/0008155138b4988190635cc600cdd11edf980b91